src/functions.rs

```diff
@@ -0,0 +1,198 @@
+        user_data: glib_sys::gpointer,
+    ) {
+        let callback: Box_<Option<Box_<dyn FnOnce() + 'static>>> =
+            Box_::from_raw(user_data as *mut _);
+        let callback = (*callback).expect("cannot get closure...");
+        callback()
+    }
+    let child_setup = if child_setup_data.is_some() {
+        Some(child_setup_func::<P> as _)
+    } else {
+        None
+    };
+    let super_callback0: Box_<Option<Box_<dyn FnOnce() + 'static>>> = child_setup_data;
+    unsafe {
+        let mut child_pid = mem::MaybeUninit::uninit();
+        let mut error = ptr::null_mut();
+        let _ = glib_sys::g_spawn_async_with_fds(
+            working_directory.as_ref().to_glib_none().0,
+            argv.to_glib_none().0,
+            envp.to_glib_none().0,
+            flags.to_glib(),
+            child_setup,
+            Box_::into_raw(super_callback0) as *mut _,
+            child_pid.as_mut_ptr(),
+            stdin_fd.as_raw_fd(),
+            stdout_fd.as_raw_fd(),
+            stderr_fd.as_raw_fd(),
+            &mut error,
+        );
+        let child_pid = from_glib(child_pid.assume_init());
+        if error.is_null() {
+            Ok(child_pid)
+        } else {
+            Err(from_glib_full(error))
+        }
+    }
+}
+
+#[cfg(any(feature = "v2_58", feature = "dox"))]
+#[cfg(windows)]
+pub fn spawn_async_with_fds<
+    P: AsRef<std::path::Path>,
+    T: AsRawHandle,
+    U: AsRawHandle,
+    V: AsRawHandle,
+>(
+    working_directory: P,
+    argv: &[&str],
+    envp: &[&str],
+    flags: SpawnFlags,
+    child_setup: Option<Box_<dyn FnOnce() + 'static>>,
+    stdin_fd: T,
+    stdout_fd: U,
+    stderr_fd: V,
+) -> Result<Pid, Error> {
+    let child_setup_data: Box_<Option<Box_<dyn FnOnce() + 'static>>> = Box_::new(child_setup);
+    unsafe extern "C" fn child_setup_func<P: AsRef<std::path::Path>>(
+        user_data: glib_sys::gpointer,
+    ) {
+        let callback: Box_<Option<Box_<dyn FnOnce() + 'static>>> =
+            Box_::from_raw(user_data as *mut _);
+        let callback = (*callback).expect("cannot get closure...");
+        callback()
+    }
+    let child_setup = if child_setup_data.is_some() {
+        Some(child_setup_func::<P> as _)
+    } else {
+        None
+    };
+    let super_callback0: Box_<Option<Box_<dyn FnOnce() + 'static>>> = child_setup_data;
+    unsafe {
+        let mut child_pid = mem::MaybeUninit::uninit();
+        let mut error = ptr::null_mut();
+        let _ = glib_sys::g_spawn_async_with_fds(
+            working_directory.as_ref().to_glib_none().0,
+            argv.to_glib_none().0,
+            envp.to_glib_none().0,
+            flags.to_glib(),
+            child_setup,
+            Box_::into_raw(super_callback0) as *mut _,
+            child_pid.as_mut_ptr(),
+            stdin_fd.as_raw_handle() as usize as _,
+            stdout_fd.as_raw_handle() as usize as _,
+            stderr_fd.as_raw_handle() as usize as _,
```

Comment:
This loses precision: usize to i32 cast. This is probably not correct and needs investigation by someone on Windows.

How do we do this elsewhere?

Reply:
We don't do that anywhere else and don't seem to have any Windows-specific fd code. Someone needs to investigate how this is supposed to work on Windows.

There's this code in GStreamer for Windows: https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/blob/90854eb0471e04a11c37fd5a5beaf7ab0cf29dfe/gstreamer/src/bus_windows.rs#L41
But I think this is equally suspicious. It's casting an `i32` to a `HANDLE` (i.e. a pointer).

I also found some code inside C GStreamer though, which does similar things and that code is actually known to work on Windows. So maybe this is all fine? Needs some investigation from someone who knows Windows :) @EPashkin ?

Reply:
Ah the part in GStreamer works because of https://github.com/gtk-rs/sys/blob/94a8fcf7119c380e27510f1bef4c546964804ef8/glib-sys/src/manual.rs#L18 . So that works all fine.

Unclear how it's possible to use these functions here on Windows though.

Reply:
I will try to look at the weekend

Reply:
Let's defer it to next release then. We said three weeks ago to not add any new features for this release anyway, and here we are nonetheless :) Let's just ignore these functions on Windows or completely for this release, and then figure it out without rushing it afterwards.

Reply:
Then let's ignore for windows. It works on unix so that'd be a shame not keep them. And yes, didn't see them when I made the regen, sorry...

Reply:
@GuillaumeGomez Then please remove windows part or add todo with comment about double checking.
Also strange than `cargo +stable fmt` passed on Appveyor but fail on travis

Reply:
I'll comment the code. About rustfmt, it's my local run. So I don't know what travis is doing...